ReactWindows/ReactNative/Modules/Storage/AsyncStorageModule.cs

```diff
@@ -20,5 +45,281 @@ public override string Name
+        {
+            Debug.Assert(keyValueArray != null);
+            Debug.Assert(callback != null);
+
+            var result = new JArray();
+            foreach (var keyValue in keyValueArray)
+            {
+                if (keyValue.Type == JTokenType.Array && keyValue.Value<JArray>().Count == 2)
+                {
+                    var pair = keyValue.Value<JArray>();
+                    if (pair.First.Type == JTokenType.String)
+                    {
+                        var key = pair.First.Value<string>();
+                        var tokenOld = getTokenFromContainer(key);
+                        var tokenNew = pair.Last;
+
+                        if (tokenOld.Type != JTokenType.Null)
+                        {
+                            if (tokenOld.Type == JTokenType.Array)
+                            {
+                                ((JArray)tokenOld).Merge(tokenNew);
+                                tokenNew = tokenOld;
+                            }
+                            else if (tokenNew.Type == JTokenType.Array)
+                            {
+                                ((JArray)tokenNew).Merge(tokenOld);
+                            }
+                            else if (tokenNew.Type == JTokenType.Null)
+                            {
+                                tokenNew = tokenOld;
+                            }
+                        }
+                        addTokenToContainer(key, tokenNew);
+                    }
+                    else
+                    {
+                        result.Add(new JArray
+                        {
+                            _invalidKey,
+                            pair.First,
+                        });
+                    }
+                }
+                else
+                {
+                    result.Add(new JArray
+                    {
+                        _invalidPair,
+                        keyValue,
+                    });
+                }
+            }
+            callback.Invoke(result);
+        }
+
+        /// <summary>
+        /// Clears the database.
+        /// </summary>
+        /// <param name="callback">Callback.</param>
+        [ReactMethod]
+        public void clear(ICallback callback)
+        {
+            Debug.Assert(callback != null);
+
+            _dataContainer.Values.Clear();
+            _typeContainer.Values.Clear();
+
+            callback.Invoke();
+        }
+
+        /// <summary>
+        /// Returns an array with all keys from the database.
+        /// </summary>
+        /// <param name="callback">Callback.</param>
+        [ReactMethod]
+        public void getAllKeys(ICallback callback)
+        {
+            Debug.Assert(callback != null);
+
+            callback.Invoke(JToken.FromObject(_dataContainer.Values.Keys)); 
+        }
+
+        /// <summary>
+        /// Adds JToken to ApplicationDataContainer with the specified key.
```

Comment:
We've been using <see cref="ApplicationDataContainer"> when referencing classes in comments.